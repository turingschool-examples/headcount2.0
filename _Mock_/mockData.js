module.exports = [
	{
		Location: 'COLORADO',
		TimeFrame: 2007,
		DataFormat: 'Percent',
		Data: 0.39465
	}
];
