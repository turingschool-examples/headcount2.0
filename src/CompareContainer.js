import React from 'react';
import CompareCard from './CompareCard'
import Card from './Card';
import { func, array} from 'prop-types'


const CompareContainer = ({ data, compareAverage, clickActive, activeCards }) => {
  // console.log(data);
  const card1 = data[0] !== undefined ?
    <Card className="comparison-card"
          clickActive={clickActive}
          city={data[0]}
          activeCards={activeCards}/> :
      null

  const card2 = data[1] !== undefined ?
    <Card clickActive={clickActive}
          city={data[1]}
          activeCards={activeCards}/> :
      null

  return (
    <div className='compare-container'>
      <CompareCard compareAverage={compareAverage}
                   data={data}
                   clickActive={clickActive}/>
      <div className='compare-card-container'>
        {card1}
        {card2}
      </div>
    </div>
  )
}

CompareContainer.propTypes = {
  data: array,
  compareAverage: func,
  clickActive: func,
  activeCards: array
}

export default CompareContainer;
