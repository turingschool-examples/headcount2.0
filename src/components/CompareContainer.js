
import React from 'react'

const CompareContainer = () => {

  //controller for managing what two cards are selected to be compared

  return (
    <div className="compare-container">
      Schools to be compared will go here
    </div>
  )
};

export default CompareContainer;
