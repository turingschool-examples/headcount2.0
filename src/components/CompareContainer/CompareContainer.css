.CompareContainer {
  width: 100%;
  background-color: #fafafa;
}