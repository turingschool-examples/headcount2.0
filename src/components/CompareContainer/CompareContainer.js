import React from 'react';
import CompareInstructions from '../CompareInstructions/CompareInstructions';
import Comparison from '../Comparison/Comparison';
import PropTypes from 'prop-types';
import './CompareContainer.css';

const CompareContainer = 
({comparisonData, handleCompareCards, removeComparison}) => {
  const getDisplay = () => {
    if (!Object.keys(comparisonData).length) {
      return; 
    } else if (Object.keys(comparisonData).length === 1) {
      return (
        <CompareInstructions 
          comparisonData={ comparisonData }
          handleCompareCards={ handleCompareCards }
          removeComparison={ removeComparison }
        />
      );
    } else {
      return (
        <Comparison
          comparisonData={ comparisonData }
          handleCompareCards={ handleCompareCards }
          removeComparison={ removeComparison }
        />
      );
    }
  };

  return (
    <section className="CompareContainer">
      {getDisplay()}
    </section>
  );
};

const { shape, string, objectOf, number, func, object } = PropTypes;

CompareContainer.propTypes = {
  handleCompareCards: func.isRequired,
  removeComparison: func.isRequired,
  comparisonData: shape({
    comparison: object
  }),
  school1: 
    objectOf(shape({
      location: string.isRequired,
      dataType: string,
      data: objectOf(
        number.isRequired)
    })),
  school2: 
    objectOf(shape({
      location: string.isRequired,
      dataType: string,
      data: objectOf(
        number.isRequired)
    }))
};

export default CompareContainer;