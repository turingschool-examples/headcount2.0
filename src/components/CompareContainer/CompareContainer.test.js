/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import CompareContainer from './CompareContainer';
import mockData from '../../data/testing_mocks';

describe('CompareContainer', () => {
  let wrapper;
  let mockedComparisonData = {};

  it('should match the snapshot if it is given an empty object', () => {
    wrapper = shallow(<CompareContainer 
                        handleCompareCards={() => {}}
                        removeComparison={() => {}}
                        comparisonData={mockedComparisonData}
                      />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if comparisonData has only 1 object', () => {
    mockedComparisonData = { 
      school1: mockData.cleanedCoKinderData
    }
    wrapper = shallow(<CompareContainer 
                        handleCompareCards={() => {}}
                        removeComparison={() => {}}
                        comparisonData={mockedComparisonData}
                      />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when it is given a full comparison', () => {
    mockedComparisonData = {
      school1: mockData.cleanedCoKinderData,
      school2: mockData.cleanedYumaKinderData,
      comparison: mockData.comparison
    }
    wrapper = shallow(<CompareContainer 
                        handleCompareCards={() => {}}
                        removeComparison={() => {}}
                        comparisonData={mockedComparisonData}
                      />)
    expect(wrapper).toMatchSnapshot();  
  })
})