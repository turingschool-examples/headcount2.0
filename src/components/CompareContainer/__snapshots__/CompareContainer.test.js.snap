// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareContainer should match the snapshot if comparisonData has only 1 object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    comparisonData={
        Object {
            "school1": Object {
              "data": Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
                "2008": 0.536,
                "2009": 0.598,
                "2010": 0.64,
                "2011": 0.672,
                "2012": 0.695,
                "2013": 0.703,
                "2014": 0.741,
              },
              "dataType": "Percent",
              "location": "COLORADO",
            },
          }
    }
    handleCompareCards={[Function]}
    removeComparison={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CompareInstructions
        comparisonData={
                Object {
                        "school1": Object {
                          "data": Object {
                            "2004": 0.24,
                            "2005": 0.278,
                            "2006": 0.337,
                            "2007": 0.395,
                            "2008": 0.536,
                            "2009": 0.598,
                            "2010": 0.64,
                            "2011": 0.672,
                            "2012": 0.695,
                            "2013": 0.703,
                            "2014": 0.741,
                          },
                          "dataType": "Percent",
                          "location": "COLORADO",
                        },
                      }
        }
        handleCompareCards={[Function]}
        removeComparison={[Function]}
/>,
      "className": "CompareContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "comparisonData": Object {
          "school1": Object {
            "data": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
            "dataType": "Percent",
            "location": "COLORADO",
          },
        },
        "handleCompareCards": [Function],
        "removeComparison": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CompareInstructions
          comparisonData={
                    Object {
                              "school1": Object {
                                "data": Object {
                                  "2004": 0.24,
                                  "2005": 0.278,
                                  "2006": 0.337,
                                  "2007": 0.395,
                                  "2008": 0.536,
                                  "2009": 0.598,
                                  "2010": 0.64,
                                  "2011": 0.672,
                                  "2012": 0.695,
                                  "2013": 0.703,
                                  "2014": 0.741,
                                },
                                "dataType": "Percent",
                                "location": "COLORADO",
                              },
                            }
          }
          handleCompareCards={[Function]}
          removeComparison={[Function]}
/>,
        "className": "CompareContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparisonData": Object {
            "school1": Object {
              "data": Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
                "2008": 0.536,
                "2009": 0.598,
                "2010": 0.64,
                "2011": 0.672,
                "2012": 0.695,
                "2013": 0.703,
                "2014": 0.741,
              },
              "dataType": "Percent",
              "location": "COLORADO",
            },
          },
          "handleCompareCards": [Function],
          "removeComparison": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CompareContainer should match the snapshot if it is given an empty object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    comparisonData={Object {}}
    handleCompareCards={[Function]}
    removeComparison={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "CompareContainer",
    },
    "ref": null,
    "rendered": null,
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "CompareContainer",
      },
      "ref": null,
      "rendered": null,
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CompareContainer should match the snapshot when it is given a full comparison 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    comparisonData={
        Object {
            "comparison": Object {
              "COLORADO": 0.53,
              "YUMA SCHOOL DISTRICT 1": 0.909,
              "compared": 0.583,
            },
            "school1": Object {
              "data": Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
                "2008": 0.536,
                "2009": 0.598,
                "2010": 0.64,
                "2011": 0.672,
                "2012": 0.695,
                "2013": 0.703,
                "2014": 0.741,
              },
              "dataType": "Percent",
              "location": "COLORADO",
            },
            "school2": Object {
              "data": Object {
                "2004": 0,
                "2005": 1,
                "2006": 1,
                "2007": 1,
                "2008": 1,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "dataType": "Percent",
              "location": "YUMA SCHOOL DISTRICT 1",
            },
          }
    }
    handleCompareCards={[Function]}
    removeComparison={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Comparison
        comparisonData={
                Object {
                        "comparison": Object {
                          "COLORADO": 0.53,
                          "YUMA SCHOOL DISTRICT 1": 0.909,
                          "compared": 0.583,
                        },
                        "school1": Object {
                          "data": Object {
                            "2004": 0.24,
                            "2005": 0.278,
                            "2006": 0.337,
                            "2007": 0.395,
                            "2008": 0.536,
                            "2009": 0.598,
                            "2010": 0.64,
                            "2011": 0.672,
                            "2012": 0.695,
                            "2013": 0.703,
                            "2014": 0.741,
                          },
                          "dataType": "Percent",
                          "location": "COLORADO",
                        },
                        "school2": Object {
                          "data": Object {
                            "2004": 0,
                            "2005": 1,
                            "2006": 1,
                            "2007": 1,
                            "2008": 1,
                            "2009": 1,
                            "2010": 1,
                            "2011": 1,
                            "2012": 1,
                            "2013": 1,
                            "2014": 1,
                          },
                          "dataType": "Percent",
                          "location": "YUMA SCHOOL DISTRICT 1",
                        },
                      }
        }
        handleCompareCards={[Function]}
        removeComparison={[Function]}
/>,
      "className": "CompareContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "comparisonData": Object {
          "comparison": Object {
            "COLORADO": 0.53,
            "YUMA SCHOOL DISTRICT 1": 0.909,
            "compared": 0.583,
          },
          "school1": Object {
            "data": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
            "dataType": "Percent",
            "location": "COLORADO",
          },
          "school2": Object {
            "data": Object {
              "2004": 0,
              "2005": 1,
              "2006": 1,
              "2007": 1,
              "2008": 1,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 1,
              "2014": 1,
            },
            "dataType": "Percent",
            "location": "YUMA SCHOOL DISTRICT 1",
          },
        },
        "handleCompareCards": [Function],
        "removeComparison": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Comparison
          comparisonData={
                    Object {
                              "comparison": Object {
                                "COLORADO": 0.53,
                                "YUMA SCHOOL DISTRICT 1": 0.909,
                                "compared": 0.583,
                              },
                              "school1": Object {
                                "data": Object {
                                  "2004": 0.24,
                                  "2005": 0.278,
                                  "2006": 0.337,
                                  "2007": 0.395,
                                  "2008": 0.536,
                                  "2009": 0.598,
                                  "2010": 0.64,
                                  "2011": 0.672,
                                  "2012": 0.695,
                                  "2013": 0.703,
                                  "2014": 0.741,
                                },
                                "dataType": "Percent",
                                "location": "COLORADO",
                              },
                              "school2": Object {
                                "data": Object {
                                  "2004": 0,
                                  "2005": 1,
                                  "2006": 1,
                                  "2007": 1,
                                  "2008": 1,
                                  "2009": 1,
                                  "2010": 1,
                                  "2011": 1,
                                  "2012": 1,
                                  "2013": 1,
                                  "2014": 1,
                                },
                                "dataType": "Percent",
                                "location": "YUMA SCHOOL DISTRICT 1",
                              },
                            }
          }
          handleCompareCards={[Function]}
          removeComparison={[Function]}
/>,
        "className": "CompareContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparisonData": Object {
            "comparison": Object {
              "COLORADO": 0.53,
              "YUMA SCHOOL DISTRICT 1": 0.909,
              "compared": 0.583,
            },
            "school1": Object {
              "data": Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
                "2008": 0.536,
                "2009": 0.598,
                "2010": 0.64,
                "2011": 0.672,
                "2012": 0.695,
                "2013": 0.703,
                "2014": 0.741,
              },
              "dataType": "Percent",
              "location": "COLORADO",
            },
            "school2": Object {
              "data": Object {
                "2004": 0,
                "2005": 1,
                "2006": 1,
                "2007": 1,
                "2008": 1,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "dataType": "Percent",
              "location": "YUMA SCHOOL DISTRICT 1",
            },
          },
          "handleCompareCards": [Function],
          "removeComparison": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
